### DISCUSSION TITLE
Act on stats

### DISCUSSION THREAD

**Discussion by intelld** [nicolargo/glances#3228] [General]
First of all thanks for a great code!

We'd like to be proactive and act upon the problems rather than just monitor them.
Question/Request:
1. How can we restart the daemon or trigger a shell script once monitored amd daemon is down?
2. How can we trigger a shell script once the monitored port or webserver RTT is long or port/web not available?
In addition, how can we collect port/web/amd stats in web mode (since --export does not work with -w)?

Thanks in advance!

**Comment by nicolargo** [comment]
Hi @intelld 

1. How can we restart the daemon or trigger a shell script once monitored AMD daemon is down ?
It is not possible for the moment. Glances AMD can call a script continuously but not if the daemon is down. This is not the purpose of Glances, have a look on monit (https://www.cyberciti.biz/tips/howto-monitor-and-restart-linux-unix-service.html) to do this kind of thinks.   

2. How can we trigger a shell script once the monitored port or webserver RTT is long or port/web not available?
Not for the moment but it is a interesting request. I open a new issue to track it and check if anothers users are interested. ==> https://github.com/nicolargo/glances/issues/1592

In addition, how can we collect port/web/amd stats in web mode (since --export does not work with -w)?
This is a limitation of the current version (see https://github.com/nicolargo/glances/issues/1527).
As a workaround, you can run two Glances instances, one for -w and one with --export.
